sdk/eventhub/azure-eventhub-checkpointstoretable/azure/eventhub/extensions/checkpointstoretable/_tablestoragecs.py

```diff
@@ -22,17 +27,217 @@ class TableCheckpointStore:
+        ownership["eventhubname"], ownership['consumergroup']),
+            u'RowKey': ownership['partitionid'],
+            u'ownerid' : ownership['ownerid'],
+        }
+        return my_new_entity
+
+    @classmethod
+    def _create_checkpoint_entity(cls, checkpoint):
+        """
+        create a dictionary with the new checkpoint attributes so that it can be updated in tables
+        """
+        my_new_entity = {
+            u'PartitionKey': "{} {} {} Checkpoint".format(checkpoint["fullyqualifiednamespace"],
+        checkpoint["eventhubname"], checkpoint['consumergroup']),
+            u'RowKey': checkpoint['partitionid'],
+            u'offset' : checkpoint['offset'],
+            u'sequencenumber' : checkpoint['sequencenumber'],}
+        return my_new_entity
+
+    def list_ownership(self, fully_qualified_namespace, eventhub_name, consumer_group):
+        """Retrieves a complete ownership list from the storage table.
+        :param str fully_qualified_namespace: The fully qualified namespace that the Event Hub belongs to.
+         The format is like "<namespace>.servicebus.windows.net".
+        :param str eventhub_name: The name of the specific Event Hub the partition ownerships are associated with,
+         relative to the Event Hubs namespace that contains it.
+        :param str consumer_group: The name of the consumer group the ownerships are associated with.
+        :rtype: Iterable[Dict[str, Any]], Iterable of dictionaries containing partition ownership information:
+                - `fully_qualified_namespace` (str): The fully qualified namespace that the Event Hub belongs to.
+                  The format is like "<namespace>.servicebus.windows.net".
+                - `eventhub_name` (str): The name of the specific Event Hub the checkpoint is associated with,
+                  relative to the Event Hubs namespace that contains it.
+                - `consumer_group` (str): The name of the consumer group the ownership are associated with.
+                - `partition_id` (str): The partition ID which the checkpoint is created for.
+                - `owner_id` (str): A UUID representing the current owner of this partition.
+                - `last_modified_time` (UTC datetime.datetime): The last time this ownership was claimed.
+                - `etag` (str): The Etag value for the last time this ownership was modified. Optional depending
+                  on storage implementation.
+        """
+        thePartitionKey = "{} {} {} Ownership".format(fully_qualified_namespace,
+        eventhub_name, consumer_group)
+        my_filter = "PartitionKey eq '"+ thePartitionKey + "'"
+        entities = self.table_client.query_entities(my_filter)
+        ownershiplist = []
+        for entity in entities:
+            dic = {}
+            dic[u'ownerid'] = entity[u'ownerid']
+            dic[u'etag'] = entity.metadata.get('etag')
+            dic[u'lastmodifiedtime'] = entity.metadata.get('timestamp')
```

Comment:
You still need to provide the `fully_qualified_namespace`, `eventhub_name`, `consumer_group` and `partition_id` for each ownership. The `partition_id` will come from the entity's row key. The other fields come from what was passed into this method.

Also, `lastmodifiedtime` should be `last_modified_time`.

I commented earlier that when storing the entity, the field names should be all lowercase with no underscores. That's separate from how ownership and checkpoints are returned to the user. You'll need to convert the entity field names back to what the ownership/checkpoint field names are.

Reply:
yea that make sense I will add the fields into the dictionary and keep it snake case